smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/WaiterGenerator.kt

```diff
@@ -0,0 +1,101 @@
+package software.amazon.smithy.swift.codegen
+
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.swift.codegen.core.CodegenContext
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.integration.SwiftIntegration
+import software.amazon.smithy.swift.codegen.model.expectShape
+import software.amazon.smithy.swift.codegen.model.hasTrait
+import software.amazon.smithy.swift.codegen.model.toLowerCamelCase
+import software.amazon.smithy.swift.codegen.utils.toUpperCamelCase
+import software.amazon.smithy.waiters.WaitableTrait
+import software.amazon.smithy.waiters.Waiter
+
+/**
+ * Generate waiters for supporting operations.  See
+ * https://smithy.io/2.0/additional-specs/waiters.html for details.
+ */
+class WaiterGenerator : SwiftIntegration {
+    override fun enabledForService(model: Model, settings: SwiftSettings): Boolean {
+        return serviceHasWaiters(model)
+    }
+
+    override fun writeAdditionalFiles(
+        ctx: CodegenContext,
+        protoCtx: ProtocolGenerator.GenerationContext,
+        delegator: SwiftDelegator
+    ) {
+        if (!serviceHasWaiters(ctx.model)) return
```

Comment:
Is this line necessary given it should be covered via the SwiftIntegration conformance `enabledForService`

Reply:
This is me being defensive... I know that `enabledForService` gets called first, then only if `true` does this method get run.  But since that logic is outside this class, I wanted this class to enforce that behavior itself.